lib/block_view.go

```diff
@@ -4557,6 +4743,102 @@ func (bav *UtxoView) _deleteProfileEntryMappings(profileEntry *ProfileEntry) {
 	bav._setProfileEntryMappings(&tombstoneProfileEntry)
 }
 
+// _getDerivedKeyMappingForOwner fetches the derived key mapping from the utxoView
+func (bav *UtxoView) _getDerivedKeyMappingForOwner(ownerPublicKey []byte, derivedPublicKey []byte) *DerivedKeyEntry {
+	// Check if the entry exists in utxoView
+	derivedKeyMapKey := MakeDerivedKeyMapKey(*NewPublicKey(ownerPublicKey), *NewPublicKey(derivedPublicKey))
+	entry, exists := bav.DerivedKeyToDerivedEntry[derivedKeyMapKey]
+	if exists {
+		return entry
+	}
+
+	// Check if the entry exists in the DB.
+	ownerPk := NewPublicKey(ownerPublicKey)
```

Comment:
Nit: Define these at the top and use them in MakeDerivedKeyMapKey

Reply:
wow... so obvious, how have I missed this one!